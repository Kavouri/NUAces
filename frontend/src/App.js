import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { LinkContainer, IndexLinkContainer } from 'react-router-bootstrap';
import { Nav, NavItem } from 'react-bootstrap';
import HomePage from './containers/HomePage';
import Add from './containers/Add';
import Profile from './containers/Profile';
import './styles/App.css';

const App = () => {
  return (
    <Router>
      <div>
        <Nav bsStyle="pills">
          <IndexLinkContainer to="/">
            <NavItem eventKey={1}> Home </NavItem>
          </IndexLinkContainer>
          <LinkContainer to="/add">
            <NavItem eventKey={2}> Add </NavItem>
          </LinkContainer>
          <LinkContainer to="/profile">
            <NavItem eventKey={3}> My Profile </NavItem>
          </LinkContainer>
        </Nav>
        <hr />
        <Route exact path="/" component={HomePage} />
        <Route path="/add" component={Add} />
        <Route path="/profile" component={Profile} />
      </div>
    </Router>
  );
};
export default App;
