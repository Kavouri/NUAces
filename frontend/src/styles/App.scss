.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

/* Please explain if this is a beginner mistake */
body, html, #root, .root {
  width: 100%;
  height: 100%;
}

.registration {
  width: 100%;
  height: 100%;
  background-image: url("https://static.pexels.com/photos/302186/pexels-photo-302186.jpeg");
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;

  .registration-form {
    background-color: rgba(255, 253, 208, 0.9);
    min-width: 15%;
    padding: 20px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    border: 1px solid black;
    .inner-registration {
      display: flex;
      flex-direction: row;
      margin: 5px 5px 5px 0px;
      justify-content: space-between;
    }
    .name-container {
    }
    .register-button {
      box-sizing: border-box;
      background-color: rgba(0, 0, 128, 0.7);
      border: none;
      border-radius: 8px;
      color: white;
      padding: 10px 24px;
      text-align: center;
      display: inline;
    }
    .register-button:hover:enabled {
      box-shadow: 0 12px 16px 0 rgba(0, 0, 0, 0.24), 0 17px 50px 0 rgba(0, 0, 0, 0.19);
    }
    .error-container {
      justify-content: center;
      color: red;
      font-style: bold;
    }
  }
}
