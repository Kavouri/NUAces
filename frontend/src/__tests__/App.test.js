import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { NavItem } from 'react-bootstrap';
import App from '../App';

let shallowApp;
let mockStore;

beforeEach(() => {
  mockStore = configureMockStore([thunk])({ partners: { partnerList: [{ name: 'partner1', id: 1 }, { name: 'partner2', id: 2 }] } });
  shallowApp = shallow(<App />);
});

describe('App', () => {
  it('renders a Nav component', () => {
    expect(shallowApp.find('Nav').length).toBe(1);
  });

  it('renders Nav Items with correct text', () => {
    const navItems = shallowApp.find('NavItem');
    expect(navItems.length).toBe(5);
    // Not sure if tests of this type are valuable.
    expect(shallowApp.contains(<NavItem eventKey={1}> Home </NavItem>)).toBe(true);
  });
  // Trying out a snapshot test
  it('matches a snapshot', () => { //failed and not debugging yet
    // eslint-disable-next-line
    const tree = renderer.create(
      <Provider store={mockStore}>
        <MemoryRouter><App /></MemoryRouter>
      </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
