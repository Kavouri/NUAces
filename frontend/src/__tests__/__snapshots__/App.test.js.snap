// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches a snapshot 1`] = `
<div>
  <ul
    className="nav nav-pills"
    role={null}
  >
    <li
      className="active"
      role="presentation"
      style={Object {}}
    >
      <a
        href="/"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
         Home 
      </a>
    </li>
    <li
      className=""
      role="presentation"
      style={undefined}
    >
      <a
        href="/add"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
         Add 
      </a>
    </li>
    <li
      className=""
      role="presentation"
      style={undefined}
    >
      <a
        href="/profile"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
         My Profile 
      </a>
    </li>
  </ul>
  <hr />
  <li>
    partner1
  </li>
  <li>
    partner2
  </li>
</div>
`;
